react/features/desktop-picker/components/DesktopPicker.js

```diff
@@ -0,0 +1,249 @@
+         */
+        dispatch: React.PropTypes.func,
+
+        /**
+         * The callback to be invoked when the component is closed or
+         * when a DesktopCapturerSource has been chosen.
+         */
+        onSourceChoose: React.PropTypes.func,
+
+        /**
+         * An object with arrays of DesktopCapturerSources. The key
+         * should be the source type.
+         */
+        sources: React.PropTypes.object,
+
+        /**
+         * Used to obtain translations.
+         */
+        t: React.PropTypes.func
+    }
+
+    /**
+     * Initializes a new DesktopPicker instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedSourceId: ''
+        };
+
+        this._onCloseModal = this._onCloseModal.bind(this);
+        this._onPreviewClick = this._onPreviewClick.bind(this);
+        this._onSubmit = this._onSubmit.bind(this);
+        this._updateSources = this._updateSources.bind(this);
+    }
+
+    /**
+     * Perform an immediate update request for DesktopCapturerSources and
+     * begin requesting updates at an interval.
+     *
+     * @inheritdoc
+     */
+    componentWillMount() {
+        this._updateSources();
+        this._startPolling();
+    }
+
+    /**
+     * Clean up component and DesktopCapturerSource store state.
+     *
+     * @inheritdoc
+     */
+    componentWillUnmount() {
+        this._stopPolling();
+        this.props.dispatch(clearDesktopSources());
+    }
+
+    /**
+     * Notifies this mounted React Component that it will receive new props.
+     * Sets a default selected source if one is not already set.
+     *
+     * @inheritdoc
+     * @param {Object} nextProps - The read-only React Component props that this
+     * instance will receive.
+     * @returns {void}
+     */
+    componentWillReceiveProps(nextProps) {
+        if (!this.state.selectedSourceId
+            && nextProps.sources.screen.length) {
+            this.setState({ selectedSourceId: nextProps.sources.screen[0].id });
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        return (
+            <Dialog
+                isModal = { false }
+                okTitleKey = 'dialog.Share'
+                onCancel = { this._onCloseModal }
+                onSubmit = { this._onSubmit }
+                titleKey = 'dialog.shareYourScreen'
+                width = 'medium' >
+                <Tabs tabs = { this._renderTabs() } />
+            </Dialog>);
+    }
+
+    /**
+     * Dispatches an action to get currently available DesktopCapturerSources.
+     *
+     * @private
+     * @returns {void}
+     */
+    _updateSources() {
+        this.props.dispatch(obtainDesktopSources(
+            typesToFetch,
+            {
+                thumbnailSize: {
```

Comment:
I thought you were only referring to the desktop source types. I've now added the thumbnail size to the config file.